pkg/tsdb/azuremonitor/azuremonitor.go

```diff
@@ -25,7 +25,7 @@ import (
 	"github.com/grafana/grafana/pkg/tsdb/azuremonitor/types"
 )
 
-func ProvideService(httpClientProvider *httpclient.Provider) *Service {
+func ProvideService() *Service {
```

Comment:
Might misunderstand, but don't you still want to get the Grafana core HTTP client provider when running as part of Grafana core?

Reply:
Based on [this](https://raintank-corp.slack.com/archives/C05QFJUHUQ6/p1708611867567199) thread we shouldn't use the core HTTP client provider as that's technically still a dependency. Also, I don't think wewant the difference in middlewares as that could get confusing😊

Reply:
yeah, no, this provider needs to stay as it is (if we don't want breaking changes). The goal would be to populate the service [here, in the standalone version,](https://github.com/grafana/grafana/blob/main/pkg/tsdb/azuremonitor/standalone/datasource.go#L20) an HTTP provider that has similar functionality to the internal one. I am still working on that, I'll let you know once there is something that can be used there.

Reply:
yes, with this change, basically all the middlewares from the [internal HTTP provider](https://github.com/grafana/grafana/blob/main/pkg/infra/httpclient/httpclientprovider/http_client_provider.go#L20) will not be executed anymore. For example, request made by Azure Monitor won't have a default User-Agent header anymore (`SetUserAgentMiddleware`).

> My opinion is that we should bring the decoupled data sources in line with how external data sources function and drop the core client if we can.

I agree, but for that we need to migrate all those middlewares to the SDK first.

Reply:
Okay cool, I will close this PR and update when you have something ready 😊